crawler/crawler-ftp/src/main/java/fr/pilato/elasticsearch/crawler/fs/crawler/ftp/FileAbstractorFTP.java

```diff
@@ -0,0 +1,215 @@
+    private final Logger logger = LogManager.getLogger(FileAbstractorFTP.class);
+
+    private FTPClient ftp;
+
+    private final OutputStream loggerOutputStream = IoBuilder.forLogger(logger).buildOutputStream();
+
+    private final PrintCommandListener ftpListener = new PrintCommandListener(new PrintWriter(loggerOutputStream));
+
+    private boolean isUtf8 = false;
+
+    private static final String ALTERNATIVE_ENCODING = "GBK";
+
+    public FileAbstractorFTP(FsSettings fsSettings) {
+        super(fsSettings);
+    }
+
+    @Override
+    public FileAbstractModel toFileAbstractModel(String path, FTPFile file) {
+        String filename = file.getName();
+        String extension = FilenameUtils.getExtension(filename);
+
+        String toEncoding = ALTERNATIVE_ENCODING;
+        if (isUtf8) {
+            toEncoding = StandardCharsets.UTF_8.displayName();
+        }
+        try {
+            filename = new String(filename.getBytes(FTP.DEFAULT_CONTROL_ENCODING), toEncoding);
+            path = new String(path.getBytes(FTP.DEFAULT_CONTROL_ENCODING), toEncoding);
+        } catch (UnsupportedEncodingException e) {
+            logger.error("Error during encoding: {}", e.getMessage());
+        }
+
+        return new FileAbstractModel(
+                filename,
+                file.isFile(),
+                // We are using here the local TimeZone as a reference. If the remote system is under another TZ, this might cause issues
+                LocalDateTime.ofInstant(Instant.ofEpochMilli(file.getTimestamp().getTimeInMillis()), ZoneId.systemDefault()),
+                // We don't have the creation date
+                null,
+                // We don't have the access date
+                null,
+                extension,
+                path,
+                path.equals("/") ? path.concat(filename) : path.concat("/").concat(filename),
+                file.getSize(),
+                file.getUser(),
+                file.getGroup(),
+                FTPUtils.getFilePermissions(file));
+    }
+
+    @Override
+    public InputStream getInputStream(FileAbstractModel file) throws IOException {
+        String fullPath = file.getFullpath();
+        if (isUtf8) {
+            fullPath = new String(fullPath.getBytes(StandardCharsets.UTF_8), FTP.DEFAULT_CONTROL_ENCODING);
+        } else {
+            fullPath = new String(fullPath.getBytes(ALTERNATIVE_ENCODING), FTP.DEFAULT_CONTROL_ENCODING);
+        }
+
+        InputStream inputStream = ftp.retrieveFileStream(fullPath);
+        if (inputStream != null) {
+            return inputStream;
+        } else {
+            throw new IOException(String.format("FTP client can not retrieve stream for [%s]", file.getFullpath()));
+        }
+    }
+
+    @Override
+    public void closeInputStream(InputStream inputStream) throws IOException {
+        inputStream.close();
+        // This is necessary if we want to read the stream
+        ftp.completePendingCommand();
```

Comment:
This line is very import for ftp crawler to retrieve multiple streams one by one. It should be placed after `inputStream.close()`, so I add `closeInputStream ` to abstract class. Also I modified related codes in `FsParserAbstract.java`